includes/tracks/class-wc-tracks-event.php

```diff
@@ -49,10 +49,17 @@ public function __construct( $event ) {
 	/**
 	 * Record Tracks event
 	 *
-	 * @return bool|WP_Error True on success, WP_Error on failure.
+	 * @return bool Always returns true.
 	 */
 	public function record() {
-		return WC_Tracks_Client::record_event( $this );
+		if (
+			( defined( 'DOING_AJAX' ) && DOING_AJAX ) ||
```

Comment:
`wp_doing_ajax()` can be used here. There isn't one for REST yet.

Reply:
Fixed in 1072d1a.